src/test/scala/com/campudus/tableaux/api/system/SystemServiceTest.scala

```diff
@@ -0,0 +1,329 @@
+
+    for {
+      emptyServices <- sendRequest("GET", "/system/services")
+    } yield {
+      assertEquals(Json.emptyArr(), emptyServices.getJsonArray("services"))
+    }
+  }
+
+  @Test
+  def retrieveAll_insertedThreeServices_returnsAllService(implicit c: TestContext): Unit = okTest {
+    val sqlConnection = SQLConnection(this.vertxAccess(), databaseConfig)
+
+    val allServiceJson = """
+                           |[
+                           |  {
+                           |    "id": 1,
+                           |    "name": "service1"
+                           |  }, {
+                           |    "id": 2,
+                           |    "name": "service2"
+                           |  }, {
+                           |    "id": 3,
+                           |    "name": "service3"
+                           |  }
+                           |]""".stripMargin
+
+    for {
+      _ <- sqlConnection.query("""
+                                 |INSERT INTO system_services
+                                 |  ("type", "name", "ordering", "active")
+                                 |VALUES
+                                 |  ('action', 'service1', 1, TRUE),
+                                 |  ('action', 'service2', 2, TRUE),
+                                 |  ('action', 'service3', 3, TRUE)
+                                 |  """.stripMargin)
+
+      allServices <- sendRequest("GET", "/system/services").map(_.getJsonArray("services"))
+    } yield {
+      assertEquals(3, allServices.size())
+      assertJSONEquals(allServiceJson, allServices.toString, JSONCompareMode.LENIENT)
+    }
+  }
+
+  // GET /system/services/<serviceId>
+
+  @Test
+  def retrieve_notExistingServices_throwsException(implicit c: TestContext): Unit =
+    exceptionTest("NOT FOUND") {
+      sendRequest("GET", "/system/services/1337")
+    }
+
+  @Test
+  def retrieve_insertedOneService_returnsService(implicit c: TestContext): Unit = okTest {
+    val sqlConnection = SQLConnection(this.vertxAccess(), databaseConfig)
+
+    val serviceJson = """
+                        |{
+                        |  "id": 1,
+                        |  "type": "action",
+                        |  "name": "service1",
+                        |  "ordering": 1,
+                        |  "displayName": {},
+                        |  "description": { "en": "english", "de": "Deutsch" },
+                        |  "active": true,
+                        |  "updatedAt": null
+                        |}""".stripMargin
+
+    for {
+      _ <- sqlConnection.query("""
+                                 |INSERT INTO system_services
+                                 |  ("type", "name", "ordering", "description", "active")
+                                 |VALUES
+                                 |  ('action', 'service1', 1, E'{"de": "Deutsch", "en": "english"}', TRUE)""".stripMargin)
```

Comment:
Wäre gut, die Werte aus dem _serviceJson_ wiederzuverwenden.

Reply:
IMHO it is better to explicitly compare strings in the test code. The DRY principle is not as important in the test code as in production and reduces the readability of the tests.